### ISSUE TITLE
LTP test wait403 fails with error `missing kernel fix`

### ISSUE THREAD

**Issue by anjalirai-intel** [gramineproject/gramine#1757]
### Description of the problem

The test checks for wait4(INT_MIN, ...) and the test fails with error ECHILD instead of ESRCH

### Steps to reproduce

Compile the wait403 test and run with gramine-direct/gramine-sgx, c code and logs has been attached in zip

[wait403.zip](https://github.com/gramineproject/gramine/files/14175152/wait403.zip)

### Expected results

TPASS: wait4 fails with ESRCH : ESRCH (3)

Summary:
passed   1
failed   0
broken   0
skipped  0
warnings 0

### Actual results

TFAIL: wait4 fails with ESRCH expected ESRCH: ECHILD (10)
[0.069] 
[0.069] HINT: You _MAY_ be missing kernel fixes:
[0.069] 
[0.069] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=dd83c161fbcc
[0.069] 
[0.069] Summary:
[0.069] passed   0
[0.069] failed   1
[0.069] broken   0
[0.069] skipped  0
[0.069] warnings 0

### Gramine commit hash

1cf1f46646646a3b9c6b371e67c80e945577456a

**Comment by dimakuv** [comment]
I confirm that this is indeed a (tiny) bug in Gramine. We do not have special casing of `INT_MIN` in the wait family of syscalls.

**Comment by mkow** [comment]
I don't see such a test in our LTP?

**Comment by dimakuv** [comment]
Yep, we have https://github.com/linux-test-project/ltp/tree/da2f34028f046a208aa2fed5e287df2538e69f91 (version `LTP 20200930`).

This test `wait403` was added in [`LTP 20210927`](https://github.com/linux-test-project/ltp/releases/tag/20210927).

So @anjalirai-intel was enabling this newer version of LTP (I think), and hit this case.
